src/main.ts

```diff
@@ -76,6 +111,22 @@ export default class VimReadingViewNavigation extends Plugin {
 		leaf.previewMode.applyScroll(scroll - this.settings.scrollDifference);
 	}
 
+    scrollTop(leaf: MarkdownView) {
+        leaf.previewMode.applyScroll(0);
+	}
+
+    scrollBottom(leaf: MarkdownView) {
+        let scroll = this.getScroll(leaf);
+        leaf.previewMode.applyScroll(scroll + this.settings.scrollDifference);
+        let newScroll = this.getScroll(leaf);
+
+        while (newScroll != scroll) {
```

Comment:
Could this end in an infinite loop?

Reply:
> Could this end in an infinite loop?

I tested it out in several cases and it doesn't seem that it does. My guess of how it works behind the scenes is that Obsidian tries to scroll down the specified about, but can't, so it scrolls to the bottom. The next iteration, it can't scroll down at all so the values coincide. If we really want a weaker condition, maybe something like
```
while (+newScroll - +scroll > 0.001)
```
but this seems ugly.

Reply:
![image](https://user-images.githubusercontent.com/83140328/183953690-f19c8770-cdcb-457e-afea-eb9825621e66.png)
Looks like you're right.

Reply:
I will admit, though, that this solutions isn't very clean. Ideally I would want to get the maximum scroll depth and directly `applyScroll`, but, after digging through the info in `leaf`, I couln't find anything.

I think this hack works for now since I didn't notice any delay after pressing `G`. What do you think?